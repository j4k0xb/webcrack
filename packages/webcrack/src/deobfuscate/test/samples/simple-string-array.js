const arr = ['log', 'Hello, World!'];
console[arr[0]](arr[1]);

// ignore mutable array
const arr2 = ['log', 'Hello, World!'];
arr2[0] = 'warn';
console[arr2[0]](arr2[1]);

const arr3 = ["requ", "m", "0x2649a392", "ire", "8", "v"];
const vm = eval(arr3[0] + arr3[3] + "(\"" + arr3[5] + "" + arr3[1] + "\")");
const v8 = eval(arr3[0] + arr3[3] + "(\"" + arr3[5] + "" + arr3[4] + "\")");

// ignore unreferenced array
const arr4 = ['log', 'Hello, World!'];