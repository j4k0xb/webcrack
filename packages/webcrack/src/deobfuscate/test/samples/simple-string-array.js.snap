console.log("Hello, World!");

// ignore mutable array
const arr2 = ["log", "Hello, World!"];
arr2[0] = "warn";
console[arr2[0]](arr2[1]);
const vm = eval("require(\"vm\")");
const v8 = eval("require(\"v8\")");

// ignore unreferenced array
const arr4 = ["log", "Hello, World!"];